### PR TITLE
Setup rack specific content rendering

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Racks now render the display content according to their real content
- Racks with only items, will render a lot of barrels on them
-

Review please

### CODE REVIEW

File: src/main/resources/assets/minecolonies/models/block/rack/blockrackfull.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+				"north": {"uv": [0, 0, 16, 16], "texture": "#11"},
+				"east": {"uv": [0, 0, 16, 16], "texture": "#11"},
+				"south": {"uv": [0, 0, 16, 16], "texture": "#11"},
+				"west": {"uv": [0, 0, 16, 16], "texture": "#11"},
+				"up": {"uv": [0, 0, 16, 16], "texture": "#11"},
+				"down": {"uv": [0, 0, 0, 0], "texture": "#11"}
+			}
+		},
+		{
+			"name": "block5",
```
Comment (Mekle001) [inline src/main/resources/assets/minecolonies/models/block/rack/blockrackfull.json:131]: nit: Duplicate Name

File: src/main/resources/assets/minecolonies/models/block/rack/blockrackfulldouble.json [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+				"north": {"uv": [0, 0, 16, 16], "texture": "#10"},
+				"east": {"uv": [0, 0, 16, 16], "texture": "#10"},
+				"south": {"uv": [0, 0, 16, 16], "texture": "#10"},
+				"west": {"uv": [0, 0, 16, 16], "texture": "#10"},
+				"up": {"uv": [0, 0, 16, 16], "texture": "#10"},
+				"down": {"uv": [0, 0, 16, 16], "texture": "#10"}
+			}
+		},
+		{
+			"name": "block5",
```
Comment (Mekle001) [inline src/main/resources/assets/minecolonies/models/block/rack/blockrackfulldouble.json:209]: Duplicate again

File: src/main/resources/assets/minecolonies/models/block/rack/blockrackfull.json (hunk 1 above)
Comment (Raycoms) [reply src/main/resources/assets/minecolonies/models/block/rack/blockrackfull.json:131]: yeah, thats the original model =D

File: src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java [hunk 3]
```diff
@@ -546,4 +579,66 @@ private void invalidateCap()
...
+     * Refresh the texture mapping.
+     */
+    private void refreshTextureCache()
+    {
+        final Map<ResourceLocation, Block> resMap = new HashMap<>();
+        int index = 0;
+        final List<Map.Entry<ItemStorage, Integer>> list = content.entrySet().stream().sorted((e1, e2) -> Integer.compare(e2.getValue(), e1.getValue())).toList();
+        for (final Map.Entry<ItemStorage,Integer> entry : list)
+        {
+            // Need more solid checks!
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java]: ?

Comment (Raycoms) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java]: old todo, removed

**Review by someaddons — APPROVED** [review]

**Pull request merged**